cmd/ethrex/ethrex.rs

```diff
@@ -497,3 +508,52 @@ fn read_known_peers(file_path: PathBuf) -> Result<Vec<Node>, serde_json::Error>
 
     serde_json::from_reader(file)
 }
+
+fn show_rich_accounts(genesis: &Genesis, contents: &str) {
+    let private_keys: Vec<String> = contents
+        .lines()
+        .filter(|line| !line.trim().is_empty())
+        .map(|line| line.trim().to_string())
+        .collect();
+
+    let mut address_to_pk = HashMap::new();
+    for pk in private_keys.iter() {
+        let pk_str = pk.strip_prefix("0x").unwrap_or(pk);
+        let Ok(pk_h256) = pk_str.parse::<H256>() else {
+            return;
+        };
+        let pk_bytes = pk_h256.as_bytes();
+        let Ok(secret_key) = SecretKey::from_slice(pk_bytes) else {
+            return;
+        };
+        let Ok(address) = get_address_from_secret_key(&secret_key) else {
+            return;
+        };
+        address_to_pk.insert(address, pk);
+    }
+
+    let mut top_accounts: Vec<(&H160, U256)> = genesis
+        .alloc
+        .iter()
+        .map(|(address, account)| (address, account.balance))
+        .collect();
+    top_accounts.sort_by(|a, b| b.1.cmp(&a.1)); // sort by greater balance
+    let number_of_top_accounts = 10;
+    top_accounts.truncate(number_of_top_accounts);
+
+    println!("Showing first {} accounts", number_of_top_accounts);
+    println!("-------------------------------------------------------------------------------");
+    for (address, balance) in top_accounts {
+        let Some(pk) = address_to_pk.get(address) else {
+            continue;
+        };
+        println!("Private Key: {}", pk);
+        println!("Address:     {:?} (Ξ {})", address, wei_to_eth(balance));
+        println!("-------------------------------------------------------------------------------");
+    }
+}
+
+fn wei_to_eth(wei: U256) -> U256 {
+    wei.checked_div(U256::from_dec_str("1000000000000000000").unwrap())
```

Comment:
What about adding underscores? Like so:
```suggestion
    wei.checked_div(U256::from_dec_str("1_000_000_000_000_000_000").unwrap())
```

I find it a bit easier to read

Reply:
I was just thinking the same thing.
I tried that, but parsing a `str` with the underscore results in a panic.